### DISCUSSION TITLE
Should we remove the "ignoreDuplicates" feature?

### DISCUSSION THREAD

**Discussion by lfcnassif** [sepinf-inc/IPED#931] [Q&A] [answered]
On #506 I removed the 'duplicate' file flag, because it isn't needed anymore to filter duplicate files by hashes.

But I kept the "ignoreDuplicates" processing feature, although I definitely don't like it, because it could ignore relevant files with same content but different metadata (like path, name or dates), like email attachments or located in p2p shared folders.

The "filter duplicates" filter on UI should be used instead of removing duplicated files from case completely beforehand, the only drawback is that larger cases (with more items) would use more computer resources (disk space, memory, cpu).

What do developers and users think about it?

**Comment by wladimirleite** [comment]
Well, removing features can always be inconvenient for users that are used to it.
But I totally agree that in general it is not a good idea to ignore duplicates at processing time, as crucial information may be lost.
So I would vote for removing it.

**Comment by lfcnassif** [comment, marked as answer]
This thread is completely related #574.

I just thought about keeping the duplicate item in case, with its hash, but zeroed content, just to keep some reference to it. But that wouldn't work for subitems, a duplicated zip/pst wouldn't be expanded. We can't get subitem references from the primary zip easily, the first zip could still be under expansion. And that could be a recursive problem... Keeping all references as items would make the analysis UI to use about the same amount of memory. There are other analysis drawbacks if just references are stored (search hits lost, filters results, etc).

Maybe one option, instead of removing it, is just putting a big WARNING in IPEDConfig about this dangerous feature.

**Comment by raulitosena** [comment]
I also would vote for removing it, if it helps on other development issues.

By keeping "filter duplicates" feature on UI, no problem for me on removing ignoreDuplicates feature (during processing time). 

In some situations, usually when examining a huge amount of pictures (millions), the UI feature ("filter duplicates") helps me mark photos of "non pertinent" category faster.

**Reply by lfcnassif** [reply]
Thanks @raulitosena, the "filter duplicates" UI feature won't be removed.

**Comment by ronaldocordeiro** [comment]
"ignoreDuplicates" option still seems to be useful when processing requires an export of selected files (for example, selected file types) and there is a concern about the size of the output data generated. Sometimes it's desirable that it fits in flash drives and duplicate files may generate extra space and require bigger storage media. Discarding duplicates may reduce the export size and allow the use of smaller flash media. Am I correct or is there any way to do that, other than using the "filter duplicates" on UI, then doing a second processing?

**Reply by lfcnassif** [reply]
IPED already reuses exported files with duplicated hashes, to not export the same content again, regardless if ignoreDuplicates is on or off.

**Reply by ronaldocordeiro** [reply]
OK. In this case, I can see no reason to keep "ignoreDuplicates" feature.

**Reply by lfcnassif** [reply]
The only advantage is processing cases faster (less items processed) and using less memory during case analysis (less items handled by UI). So, to not make users of that option sad, I decided to not remove it and put a big WARNING in IPEDConfig about its inherent problems.
